feat: deprecate test mode and never frequency

jeffersonrabb opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

### Changes proposed in this Pull Request:

Test with https://github.com/Automattic/newspack-plugin/pull/809.

Replaces https://github.com/Automattic/newspack-popups/pull/365

### How to test the changes in this Pull Request:

Full testing instructions in https://github.com/Automattic/newspack-plugin/pull/809

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

adekbadek reviewed (changes requested):
As for the migration strategy, WDYT about simply 
1. marking `test` & `never` popups as inactive [in the wizard](https://github.com/Automattic/newspack-plugin/pull/809)
2. filtering these so they are never shown to a reader
?

jeffersonrabb commented on includes/class-newspack-popups.php:
```diff
@@ -477,24 +478,41 @@ public static function popup_default_fields( $post_id, $post, $update ) {
+			case 'above-header':
+			case 'manual':
+			default:
+				$dismiss_text = null;
```
Inline campaigns generally don't need permanent dismissal buttons. That feature is really only required for intrusive campaigns like pop-ups.

adekbadek replied:
True, I forgot that an empty message will hide the button 👍

dkoo commented:
> As for the migration strategy, WDYT about simply
> 
> 1. marking `test` & `never` popups as inactive [in the wizard](https://github.com/Automattic/newspack-plugin/pull/809)
> 2. filtering these so they are never shown to a reader
>    ?

Since we've simplified campaign statuses to published vs. unpublished, what if we were to programmatically unpublish any "test" and "never" campaigns that are found and treat them the same as draft campaigns?

jeffersonrabb commented on includes/class-newspack-popups-inserter.php:
```diff
@@ -676,13 +653,11 @@ public static function should_display( $popup, $skip_context_checks = false ) {
 		if ( $general_conditions && Newspack_Popups_View_As::viewing_as_spec() ) {
-			return $is_not_test_mode;
+			return true;
 		}
```
Done in https://github.com/Automattic/newspack-popups/pull/390/commits/6ef265b9fa6d061d1f421a7e70f72e94aa2fcf2d

jeffersonrabb commented on src/editor/StatusSidebar.js:
```diff
@@ -7,55 +7,16 @@
-						'newspack-popups'
-					) }
-					onChange={ value => onMetaFieldChange( 'frequency', value ? 'test' : 'once' ) }
-				/>
```
Fixed in https://github.com/Automattic/newspack-popups/pull/390/commits/a4ee3e057c5f1793c1bb2a28e0c8a5061f4ad75b

jeffersonrabb commented:
> Since we've simplified campaign statuses to published vs. unpublished, what if we were to programmatically unpublish any "test" and "never" campaigns that are found and treat them the same as draft campaigns?
> 
> Edit to add: On the front-end, we should still filter any `test` or `never` campaigns so they're never shown, as a failsafe.

This is now the case, implemented (mostly) in https://github.com/Automattic/newspack-popups/pull/390/commits/72f99434eaa149416b73ec55762127fd7e8bba33. Complete explanation and testing instructions in https://github.com/Automattic/newspack-plugin/pull/809.

jeffersonrabb commented on includes/class-newspack-popups-inserter.php:
```diff
@@ -676,25 +653,20 @@ public static function should_display( $popup, $skip_context_checks = false ) {
-			return $general_conditions && $is_not_test_mode;
+			return $general_conditions;
 		}
 		// Hide non-test mode campaigns for logged-in users.
```
Fixed in https://github.com/Automattic/newspack-popups/pull/390/commits/c8c76eae3e3649ff428969424ca069fde8fa2a2e

jeffersonrabb commented on src/editor/StatusSidebar.js:
```diff
@@ -7,65 +7,26 @@
-					onChange={ value => onMetaFieldChange( 'frequency', value ? 'test' : 'once' ) }
-				/>
-				{ isOverlay && (
+		isOverlay && (
```
Fixed in https://github.com/Automattic/newspack-popups/pull/390/commits/c8c76eae3e3649ff428969424ca069fde8fa2a2e

dkoo reviewed (approved):
The migration works great and the test/never campaigns are properly unpublished and don't appear in preview or on the front-end.
